platforms/jvm/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/junit/JUnitTestClassExecutor.java

```diff
@@ -115,11 +116,14 @@ private void runTestClass(String testClassName) throws ClassNotFoundException {
 
         if (spec.isDryRun()) {
             runner = new JUnitTestDryRunner(runner);
```

Comment:
💭 It might be clearer to declare a new variable here to make it clear what `Runner` is being used.

Reply:
This is actually reworked in a subsequent PR, so I'm going to leave it here and it will be cleaned up afterwards.